cmd/nephe-controller/main.go

```diff
@@ -47,6 +56,124 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+// Secret validator
+type secretValidator struct {
+	client  controllerclient.Client
+	decoder *admission.Decoder
+}
+
+// allowSecretUpdate returns nil only when secret name/namespace/key are same.
```

Comment:
create a separate file for it.. Dont add in main

Reply:
Done.